### PR TITLE
Fix issue #7434 - nessus_db_scan_workspace: wrong number of arguments

### PR DESCRIPTION
**Pull request by dhn**
## What this does

This piece of code fix my issue #7434. 

## Verification

List the steps needed to make sure this thing works

- [x] Start `msfconsole`
- [x] `load nessus`
- [x] `nessus_connect  USER:PASSWORD@IP:8834`
- [x] `nessus_db_scan_workspace <policy ID> <scan name> <scan description> <workspace>`

## Example Output

```
msf > load nessus
[*] Nessus Bridge for Metasploit
[*] Type nessus_help for a command listing
[*] Successfully loaded plugin: Nessus

msf > nessus_connect  msf:pass@10.0.2.15:8834
[*] Connecting to https://10.0.2.15:8834/ as msf
[*] User msf authenticated successfully.

msf > workspace testlab
[*] Workspace: testlab

msf > workspace 
  default
* testlab

msf > services 

Services
========

host       port  proto  name        state  info
----       ----  -----  ----        -----  ----
127.0.0.1  22    tcp    ssh         open   
127.0.0.1  111   tcp    rpcbind     open   
127.0.0.1  5432  tcp    postgresql  open   
127.0.0.5  22    tcp    ssh         open   
127.0.0.5  111   tcp    rpcbind     open   

msf > nessus_policy_list 
Policy ID  Name     Policy UUID
---------  ----     -----------
4          advance  ad629e16-03b6-8c1d-cef6-ef8c9dd3c658d24bd260ef5f9e66


msf > nessus_db_scan_workspace  ad629e16-03b6-8c1d-cef6-ef8c9dd3c658d24bd260ef5f9e66 "Nessus Testscan" "only a test scan" testlab
[*] Switched workspace: testlab
[*] Targets: 127.0.0.1,
[*] Targets: 127.0.0.1,127.0.0.5,
[*] Creating scan from policy ad629e16-03b6-8c1d-cef6-ef8c9dd3c658d24bd260ef5f9e66, called "Nessus Testscan" and scanning all hosts in testlab
[*] Scan ID 56 successfully created
[*] Run nessus_scan_launch 56 to launch the scan

msf > nessus_scan_launch 56
[+] Scan ID 56 successfully launched. The Scan UUID is 2ac6fd7e-2005-3ae4-88fd-eaf857f8f10c8fb6a794084359a6

msf > 

```

### CODE REVIEW

File: plugins/nessus.rb [hunk 1]
```diff
@@ -1136,7 +1136,14 @@ def cmd_nessus_db_scan_workspace(*args)
         end
         targets.chop!
         print_status("Creating scan from policy #{policy_id}, called \"#{name}\" and scanning all hosts in #{framework.db.workspace.name}")
-        scan = @n.scan_create(policy_id, name, desc, targets)
+        et=Hash.new
```
Comment (wvu-r7) [inline plugins/nessus.rb]: How about creating the hash normally? Like ```et = { 'enabled' => false, [...] }```.

Comment (dhn) [inline plugins/nessus.rb]: I've used only the same syntax as in `cmd_nessus_scan_new` and `cmd_nessus_db_scan` ;-).

```
root@kali:/usr/share/metasploit-framework/plugins# grep -n -A2 "et=Hash.new" nessus.rb 
971:          et=Hash.new
972-          et['enabled']=false
973-          et['launch']='ONETIME'
--
1080:        et=Hash.new
1081-        et['enabled']=false
1082-        et['launch']='ONETIME'
--
1139:        et=Hash.new
1140-        et['enabled']=false
1141-        et['launch']='ONETIME'
root@kali:/usr/share/metasploit-framework/plugins# 

```

Comment (wvu-r7) [inline plugins/nessus.rb]: Yuck!

**Review by dhn — COMMENTED** [review]
I've refactored the bad hash creation notation in nessus.rb, so now i hope it looks better

**Comment by wvu-r7** [comment]
You rock. :D

**Comment by busterb** [comment]
Had my eye on this one, looks good.

**Comment by busterb** [comment]
## Release Notes

This fixes a bug in the nessus plugin where a user-invoked scan via the `nessus_db_scan_workspace` would display an error.

**Pull request merged**